index.js

```diff
@@ -44,8 +44,11 @@ module.exports.pitch = function (request) {
 		"// Prepare cssTransformation",
 		"var transform;",
 		options.transform ? "transform = require(" + loaderUtils.stringifyRequest(this, "!" + path.resolve(options.transform)) + ");" : "",
-		"var options = " + JSON.stringify(options),
-		"options.transform = transform",
+                "var insertIntoCallable;",
+                options.insertIntoCallable ? "insertIntoCallable = require(" + loaderUtils.stringifyRequest(this, "!" + path.resolve(options.insertIntoCallable)) + ")" : "",
```

Comment:
This currently introduces a new options to the loader (`options.insertIntoCallable`) ?  I would be much in favor to resolve this via type checking of `options.insertInto` e.g `typeof options.InsertInto === 'function'` instead of adding yet another option to the loader. Passing a `{Function}` is a good idea I planned to reduce the `options`  of `style-loader` whenever possible in a similar way  for the next major release since most of current `options` are kind of `style-loader` specific 'weirdness'  due to technical debt and legacy support in the code.

Reply:
Yeah, first time I'm trying to use existing options but loader have validation by schema and this is validation check for string. At second there is should be a path to file. As we can't use function directly we can't check if with typeof operator and have 100% guarantee that given string is path to file

Reply:
Another idea: what if use `.toString()` for function and then eval it. Not sure how it is safe

Reply:
> Yeah, first time I'm trying to use existing options but loader have validation by schema and this is validation check for string.

This can be fixed as the `options` validation is able to handle that via `oneOf` && `typeof: 'function'` now (the dependency in `style-loader` likely needs an update). Simply remove the the check for `insertInto` while developing for now and if needed I will help/take over at this point then 

> As we can't use function directly we can't check it with typeof operator and have 100% guarantee that given string is path to file

**webpack.config.js**
```js
{
  insertInto: require('path/to/custom/targetFn') // {Object}
  insertInto () { // {Function}
    return document.querySelector('#foo')
 }
}
```

**index.js [L16+](https://github.com/savelichalex/style-loader/blob/7902ef3aafede882471fdb0b03d968541965b7e3/index.js#L16)**

```js
// handler (within the loader phase)
const insertInto = typeof options.insertInto === 'function' ? ...
...
// pass the result to code block
```

> Another idea: what if use .toString() for function and then eval (or inline) it. Not sure how it is safe

You can certainly try go in that direction since `webpack` will parse this code again so a function declaration as {String} should work (will be/should be 'evaluated' or parsed as `{Function}`)

Reply:
> This can be fixed as the options validation is able to handle that via oneOf && typeof: 'function' now

Looks like `oneOf` in `ajv` not help, need to define new keyword